### PR TITLE
bitbucket login

### LINKED ISSUE #4: [Feature Request]: Bitbucket Login
### Preflight Checklist

- [X] I have read the Contributions.md for this project.
- [X] I agree to follow the Code of Conduct that this project adheres to.
- [X] I have searched the feature for a feature request that matches the one I want to file, without success.

### Problem Description

Bitbucket Login doesn't exist but many open source softwares are still maintained on Bitbucket.
Thus Bitbucket should be supported.

### Proposed Solution

We can use `passport-bitbucket-oauth2` as we are already using `passport`

### Alternatives Considered

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by amir-ammar**
# Description:
bitbucket login has been implemented so, now you can login with your bitbucket account
# Packages
- passport
- passport-bitbucket-oauth2

Fixes #4 

## Type of change:

-   [x] New feature (bitbucket login)

# Checklist:
-   [x] My code follows the style guidelines of this project.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: sample.env [hunk 1]
```diff
@@ -1,12 +0,0 @@
-MONGO_DB_URI=Xx
```
Comment (Shubham-Kumar-2000) [inline sample.env:1]: Please update the `sample.env` properly

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: sample.env (hunk 1 above)
Comment (amir-ammar) [reply sample.env:1]: okey

File: sample.env [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
-
-GITHUB_CLIENT_ID=XX
-GITHUB_CLIENT_SECRET=XX
+MONGO_DB_URI=XX
+
+PORT=3000
+
+CLUSTER=yes
+
+BASE_URL=http://localhost:3000
```
Comment (Shubham-Kumar-2000) [inline sample.env]:
```
PORT=4000

CLUSTER=yes

BASE_URL=http://localhost:4000
FRONTEND_URL=http://localhost:3000
```

Make default backend to port `4000` and frontend to `3000`

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: sample.env (hunk 2 above)
Comment (amir-ammar) [reply sample.env]: Done

File: sample.env [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
-
-BASE_URL=http://localhost:4000
-FRONTEND_URL=http://localhost:3000
-TOKEN_SECRET=XX
-
-GITHUB_CLIENT_ID=XX
-GITHUB_CLIENT_SECRET=XX
+MONGO_DB_URI=XX
+
+PORT=3000
```
Comment (Shubham-Kumar-2000) [inline sample.env]: Port of `base_url` and Port are not matching

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: sample.env (hunk 3 above)
Comment (amir-ammar) [reply sample.env]: I have done it, sorry for these lot of problem  , because this the first time to contribute to open source

**Review by Shubham-Kumar-2000 — APPROVED** [review]

**Pull request merged**